sdk/python/kfp/_client.py

```diff
@@ -1137,7 +1139,8 @@ def upload_pipeline_version(
     response = self._upload_api.upload_pipeline_version(
       pipeline_package_path,
       name=pipeline_version_name,
-      pipelineid=pipeline_id
+      pipelineid=pipeline_id,
+      description=description,
```

Comment:
if we are using an old version of kfp-server-api, will passing None to this `description` field break, because there is no `description` field?

Reply:
Given the code, I don't think so.  It will be a noop really. It may also be reasonable to ask users to upgrade the server-api to use this as otherwise its pretty much not usable.

https://github.com/kubeflow/pipelines/blob/master/backend/api/python_http_client/kfp_server_api/api/pipeline_upload_service_api.py#L138

Reply:
As reported in https://github.com/kubeflow/pipelines/issues/6256#issuecomment-908025917, this is actually a breaking change. I'd suggest constructing args to upload_pipeline_version using a dictionary, and then passing the dictionary to the method. In this way, when user does not provide their own description, we can omit the description key altogether.

Reply:
You are right, I missed the validation step on line 121. Opened #6472 to do the approach. Let's discuss there how to best handle this